src/DotLiquid.Tests/FilterTests.cs

```diff
@@ -206,6 +267,111 @@ public void TestFilterWithMultipleMethodSignaturesAndContextParam()
+                localFilters: new[] { typeof(FiltersWithMultipleMethodSignaturesDifferentClassesOne), typeof(FiltersWithMultipleMethodSignaturesDifferentClassesTwo) });
+        }
+
+
+        [Test]
+        public void TestFilterWithMultipleMethodSignaturesAndContextParamInDifferentClasses()
+        {
+            Template.RegisterFilter(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne));
+            Template.RegisterFilter(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo));
+
+            Assert.AreEqual("AB", Template.Parse("{{'A' | concat_with_context : 'B'}}").Render());
+            Assert.AreEqual("ABC", Template.Parse("{{'A' | concat_with_context : 'B', 'C'}}").Render());
+        }
+
+        [Test]
+        public void TestFilterAsLocalFilterWithMultipleMethodSignaturesAndContextDifferentClasses()
+        {
+
+            Helper.AssertTemplateResult(
+                expected: "AB // ABC",
+                template: "{{'A' | concat_with_context : 'B'}} // {{'A' | concat_with_context : 'B', 'C'}}",
+                localVariables: null,
+                localFilters: new[] { typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne), typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo) });
+        }
+
+
+        [Test]
+        public void TestFilterInContextWithMultipleMethodSignaturesAndContextParamInDifferentClasses()
+        {
+            _context.AddFilters(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne));
+            _context.AddFilters(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo));
+
+            Assert.AreEqual("AB", new Variable("'A' | concat_with_context : 'B'").Render(_context));
+            Assert.AreEqual("ABC", new Variable("'A' | concat_with_context : 'B', 'C'").Render(_context));
+        }
+
+        [Test]
+        public void TestFilterOverridesMethodWithSameMethodSignaturesDifferentClasses()
+        {
+            Template.RegisterFilter(typeof(FilterWithSameMethodSignatureDifferentClassOne));
+            Template.RegisterFilter(typeof(FilterWithSameMethodSignatureDifferentClassTwo));
+
+            Assert.AreEqual("ABClass Two", Template.Parse("{{'A' | concatenate : 'B'}}").Render());
+            Assert.AreNotEqual("ABClass One", Template.Parse("{{'A' | concatenate : 'B'}}").Render());
+        }
+
+        [Test]
+        public void TestFilterInContextOverridesMethodWithSameMethodSignaturesDifferentClasses()
+        {
+            _context.AddFilters(typeof(FilterWithSameMethodSignatureDifferentClassOne));
+            _context.AddFilters(typeof(FilterWithSameMethodSignatureDifferentClassTwo));
+
+            Assert.AreEqual("ABClass Two", new Variable("'A' | concatenate : 'B'").Render(_context));
+            Assert.AreNotEqual("ABClass One", new Variable("'A' | concatenate : 'B'").Render(_context));

```

Comment:
It would be wise to have at least one test case which ends up calling class one so we are not picking class two for another unintended reason that filter add order.

Reply:
updated: https://github.com/dotliquid/dotliquid/pull/453/commits/fc59073b69e63abae0e66b343c938f05d0d15e30